### PR TITLE
PWA: workaround Firefox not being able to stream DOM

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: src/shadow-embed.js [hunk 1]
```diff
@@ -494,3 +595,100 @@ export class ShadowDomWriter {
...
+  }
+
+  /** @override */
+  write(chunk) {
+    dev().assert(this.onBody_ && this.onBodyChunk_ && this.onEnd_);
+    if (this.eof_) {
+      throw new Error('closed already');
+    }
+    if (chunk) {
+      this.fullHtml_ += dev().assertString(chunk);
```
Comment (jridgewell) [inline src/shadow-embed.js]: Nit: use an array, and join it at the end.

Comment (dvoytenko) [inline src/shadow-embed.js]: Good point. Done.

Comment (cramforce) [inline src/shadow-embed.js]: They should be effectively the same speed.

**Comment by rsimha-amp** [comment]
@dvoytenko, I'm seeing the following error message when I do `gulp build` on a clean client. Looks like it has to do with this PR.

```
/usr/local/google/home/rsimha/src/amphtml/src/shadow-embed.js: Line 21: Duplicate declaration "vsyncFor" while parsing file: /usr/local/google/home/rsimha/src/amphtml/src/shadow-embed.js
```

**Pull request merged**